ci/docker/vt.dockerfile

```
@@ -0,0 +1,46 @@
+ARG REPO=lifflander1/vt
+ARG ARCH=amd64
+ARG IMAGE=wf-amd64-ubuntu-22.04-gcc-12-cpp
+
+ARG BASE=${REPO}:${IMAGE}
+
+FROM --platform=${ARCH} ${BASE} AS build
+
+ARG BUILD_SHARED_LIBS
+ARG CMAKE_BUILD_TYPE
+ARG CMAKE_CXX_STANDARD
+ARG VT_ASAN_ENABLED
+ARG VT_BUILD_TRACE_ONLY
+ARG VT_CI_BUILD
+ARG VT_CI_TEST_LB_SCHEMA
+ARG VT_CODE_COVERAGE
+ARG VT_DEBUG_VERBOSE
+ARG VT_DIAGNOSTICS_ENABLED
+ARG VT_DIAGNOSTICS_RUNTIME_ENABLED
+ARG VT_DOXYGEN_ENABLED
+ARG VT_EXTENDED_TESTS_ENABLED
+ARG VT_EXTERNAL_FMT
+ARG VT_FCONTEXT_ENABLED
+ARG VT_INCLUSION_TYPE
+ARG VT_KOKKOS_ENABLED
+ARG VT_LB_ENABLED
+ARG VT_MIMALLOC_ENABLED
+ARG VT_MPI_GUARD_ENABLED
+ARG VT_NO_COLOR_ENABLED
+ARG VT_PERF_ENABLED
+ARG VT_POOL_ENABLED
+ARG VT_PRODUCTION_BUILD_ENABLED
+ARG VT_RDMA_TESTS_ENABLED
+ARG VT_TESTS_NUM_NODES
+ARG VT_TRACE_ENABLED
+ARG VT_TRACE_RUNTIME_ENABLED
+ARG VT_TV_ENABLED
+ARG VT_UBSAN_ENABLED
+ARG VT_UNITY_BUILD_ENABLED
+ARG VT_WERROR_ENABLED
+ARG VT_ZOLTAN_ENABLED
+
+RUN --mount=target=/vt \
```

My thinking is that we should use something like this for caching:
```suggestion
RUN --mount=target=/vt --mount=type=cache,source=/ccache,target=/build/ccache \
```
(in combination with the usual `actions/cache@v4`).
Unfortunately I haven't been able to get it to work yet.